packages/gatsby-theme-aio/src/components/Frame/index.js

```diff
@@ -34,86 +36,104 @@ const Frame = ({ src, height = 'calc(100vh - var(--spectrum-global-dimension-siz
     }
   }, [location.pathname]);
 
+  useEffect(() => {
+    setConnectionReady(false);
+    if (iframe != null && ims != null) {
```

Comment:
Is it OK to assume that the ims provided through context will always eventually be non-null, and prevent the iframe from rendering if it remains null? @icaraps

Reply:
No the iframe should load regardless of the IMS state

Reply:
But I think your point is that IMS should be ready before calling the IMS specific functions right ?

Reply:
Ok, I'll have to find a workaround. I had an issue where the child was only seeing null values for ims (I think the value of ims gets captured at the time when createConnection is called) even after the parent received a non-null value.

Reply:
I removed the null check, I couldn't reproduce the bug I was seeing earlier even with set timeout. Must have been caused by something else.

Reply:
@hannessolo sorry to resurrect an old thing but I'm noticing the same behaviour where the ims is null around half the time for me. Did you ever figure out a work around?

Reply:
I can't recall whether I did anything to fix it, sorry - it does work reliably in the app I was working on though.

You can find the code that consumes the IMS orgs here, maybe it helps: 

https://git.corp.adobe.com/IODevEnablement/app-builder-trial-form/blob/main/logic/IMSOrgProvider.ts#L29
https://git.corp.adobe.com/IODevEnablement/app-builder-trial-form/blob/9e4d26fca1edea049d84bfdf659ac2d1430e6b7a/pages/index.tsx#L24
https://git.corp.adobe.com/IODevEnablement/app-builder-trial-form/blob/9e4d26fca1edea049d84bfdf659ac2d1430e6b7a/slides/DetailsSlide.tsx#L42

Does IMS stay null, or does it eventually recieve a value? Ie. if you delay calling the `connectToParent` in the child app by 500ms, is it still null? I remember having timing issues but I thought this PR should have fixed them.

Reply:
It seems that ims stays null but I don't have access to the inner frame code to check if actually ever becomes not null. I suspect when the `connectToParent` calls, it just fires a little too early before IMS inits and does its thing. 

I'll have a look at those code samples though and see. Thanks for the response!